src/proj_4D_api.c

```diff
@@ -519,6 +519,19 @@ Returns 1 on success, 0 on failure
     if (P->is_geocent || P->helmert || do_cart) {
         char def[150];
         sprintf (def, "break_cs2cs_recursion     proj=cart   a=%40.20g  es=%40.20g", P->a_orig, P->es_orig);
+        {
```

Comment:
This block is stylistically quite different from its surroundings. I would prefer something like this (untested):
```
if (P->is_geocent || P->helmert || do_cart) {
    char def[150];
    char *next_pos = def;

    sprintf (def, "break_cs2cs_recursion     proj=cart   a=%40.20g  es=%40.20g", P->a_orig, P->es_orig);

    /* In case the current locale does not use dot but comma as decimal */
    /* separator, replace it with dot, so that proj_atof() behaves */
    /* correctly. */
    /* TODO later: use C++ ostringstream with imbue(std::locale::classic()) */
    /* to be locale unaware */
    while ( (next_pos =  strchr (next_pos, ',')))
        *next_pos++ = '.';

    ...
```
or even a for rather than while:

    for (next_pos = def; next_pos = strchr (next_pos, ','); next_pos++)
        *next_pos = '.'

Alternatively, we could build comma-dot agnosticism into the alternative proj_strtod in proj_strtod.c